src/sql/src/plan/statement/ddl.rs

```diff
@@ -395,15 +395,26 @@ pub fn plan_create_source(
                         _ => sql_bail!("{} is not a kafka connection", item.name()),
                     };
 
-                    if !with_options.is_empty() {
+                    if with_options
+                        .iter()
+                        .filter(|opt| {
```

Comment:
I think an `any` pattern here is a little cleaner

Reply:
Can you also definitely throw a comment here explaining what's up--i.e. these are special `WITH` options that we must support for some really compelling reasons